state-surgery/util.go

```diff
@@ -0,0 +1,22 @@
+package state_surgery
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+func wrapErr(err error, msg string, ctx ...any) error {
+	return fmt.Errorf("%s: %v", fmt.Sprintf(msg, ctx...), err)
```

Comment:
```suggestion
	return fmt.Errorf("%s: %w", fmt.Sprintf(msg, ctx...), err)
```

nit: Using `%w` enables checking nested errors. Not that important here, but something that I think we should be using more across our go code.